modules/core/src/main/java/org/apache/ignite/internal/commandline/cache/CheckIndexInlineSizes.java

```diff
@@ -0,0 +1,176 @@
+        Objects.equals(node.attribute(ATTR_CLIENT_MODE), false) && !"true".equals(node.attribute(ATTR_DAEMON));
+
+    /** {@inheritDoc} */
+    @Override public Object execute(GridClientConfiguration clientCfg, Logger log) throws Exception {
+        try (GridClient client = Command.startClient(clientCfg)) {
+            Set<GridClientNode> serverNodes = client.compute().nodes().stream()
+                .filter(SRV_NODES)
+                .collect(toSet());
+
+            Set<GridClientNode> supportedServerNodes = serverNodes.stream()
+                .filter(n -> checkIndexInlineSizesSupported(n))
+                .collect(toSet());
+
+            Collection<UUID> serverNodeIds = F.transform(serverNodes, GridClientNode::nodeId);
+            Collection<UUID> supportedServerNodeIds = F.transform(supportedServerNodes, GridClientNode::nodeId);
+
+            CheckIndexInlineSizesResult res = client.compute().projection(supportedServerNodes).execute(
+                CheckIndexInlineSizesTask.class.getName(),
+                new VisorTaskArgument<>(supportedServerNodeIds, false)
+            );
+
+            Set<UUID> unsupportedNodes = serverNodeIds.stream()
+                .filter(n -> !supportedServerNodeIds.contains(n))
+                .collect(toSet());
+
+            analizeResults(log, unsupportedNodes, res);
+        }
+
+        return null;
+    }
+
+    /** */
+    private void analizeResults(
+        Logger log,
+        Set<UUID> unsupportedNodes,
+        CheckIndexInlineSizesResult res
+    ) {
+        if (!F.isEmpty(unsupportedNodes))
+            log.info(String.format(NOT_ALL_NODES_SUPPORT_FEATURE_WARN_MSG_FMT, unsupportedNodes));
+
+        Map<String, Map<Integer, Set<UUID>>> indexToSizeNode = new HashMap<>();
+
+        for (Map.Entry<UUID, Map<String, Integer>> nodeRes : res.inlineSizes().entrySet()) {
+            for (Map.Entry<String, Integer> index : nodeRes.getValue().entrySet()) {
+                Map<Integer, Set<UUID>> sizeToNodes = indexToSizeNode.computeIfAbsent(index.getKey(), x -> new HashMap<>());
+
+                sizeToNodes.computeIfAbsent(index.getValue(), x -> new HashSet<>()).add(nodeRes.getKey());
+            }
+        }
+
+        log.info("Found " + indexToSizeNode.size() + " secondary indexes.");
+
+        Map<String, Map<Integer, Set<UUID>>> problems = indexToSizeNode.entrySet().stream()
+            .filter(e -> e.getValue().size() > 1)
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        if (F.isEmpty(problems))
+            log.info(INDEXES_INLINE_SIZE_ARE_SAME);
+        else
+            printProblemsAndShowRecommendations(problems, log);
+    }
+
+    /** */
+    private void printProblemsAndShowRecommendations(Map<String, Map<Integer, Set<UUID>>> problems, Logger log) {
+        log.info(problems.size() +
+            " indexes have different effective inline size on nodes. It can lead to performance degradation in SQL queries.");
```

Comment:
and what if it only one index ?

Reply:
Do you mean a plural form of index**es**?

Reply:
yep )